### PR TITLE
NO-ISSUE: Soft install timeout enhancement

### PR DESCRIPTION
**Pull request by ori-amizur**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md
/cc @carbonin 
/cc @gamli75

### CODE REVIEW

File: docs/enhancements/soft-install-timeout.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+failed install, I was able to easily resolve the issue and get the cluster
+operator to report success. But the service no longer cared; as far as it was
+concerned, the installation failed and was permanently marked as such. It also
+would not give me the kubeadmin password, which is an important feature of the
+install experience and tricky to obtain otherwise. I would like the service
+inform me about the exceeded timeout, but it should give me the kubeadmin
+password (if available at that point) and after my fixes it should continue
+and eventually mark the installation as successful.
+
+#### Allow manually fixing a late binding installation
```
Comment (carbonin) [inline docs/enhancements/soft-install-timeout.md:57]: I'm not sure what late binding has to do with this feature.

Are we explicitly disabling the soft-timeouts for "early" binding even if the feature is enabled?

File: docs/enhancements/soft-install-timeout.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+In the flow managed by the controller all steps must be completed (in the table above).  The last step is a 
+a notification to complete installation.
+
+### Suggested changes
+- Only assisted service should notify installation completion
+- All steps by controller (as specified in the above table) will be known as stages in the assisted service.  This will enable the service to manage 
+them in a similar way that host stages are managed.
+- Controller will not stop activity due to timeout.
+- Assisted service should be able to terminate the controller.
+- All existing timeouts except installation timeout and rebooting timeout should be treated as soft timeouts (i.e cluster or host events)
```
Comment (carbonin) [inline docs/enhancements/soft-install-timeout.md:124]: What happens if these timeouts are exceeded in this case?
The cluster is still moved to error?

File: docs/enhancements/soft-install-timeout.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+- Since OLM operators are not mandatory for successful installation, in case of timeout on these operators, the installation will proceed and marked as successul.
+- The suggested functionality should be optional. For SaaS it will be enabled globally by default and will be disabled by default
+for ZTP.  In addition, to use this feature it has to be enabled at organizational level. 
+- The 24h global installation timeout will be kept.  It will be considered as hard timeout. 
+
+### Open Questions
+
+- Should we collect must gather logs when the installation timeout expires, even
+  if we don't mark it as failed?
```
Comment (carbonin) [inline docs/enhancements/soft-install-timeout.md:133]: What was the resolution here?

File: docs/enhancements/soft-install-timeout.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+>
+> Most installations complete in approximately 45 minutes, but it took 23 hours
+> and 52 minutes already. Check the logs to find out why or reset the cluster
+> to start over.
+
+The progress bars and other UI elements used to indicate progress should also
+explicitly indicate that the installation is taking longer than expected, for
+example using warning icons or specific colors. For example:
+
+![UI example](./soft-install-timeout/ui-example.png)
```
Comment (carbonin) [inline docs/enhancements/soft-install-timeout.md:151]: Was this done for the CIM UI?

@rawagner ?

File: docs/enhancements/soft-install-timeout.md (hunk 1 above)
Comment (ori-amizur) [reply docs/enhancements/soft-install-timeout.md:57]: I agree.  Will remove the paragraph

File: docs/enhancements/soft-install-timeout.md (hunk 2 above)
Comment (ori-amizur) [reply docs/enhancements/soft-install-timeout.md:124]: The cluster will move to error due to timeout only on the hard 24h timeout.
All the rest are soft timeouts.

File: docs/enhancements/soft-install-timeout.md (hunk 3 above)
Comment (ori-amizur) [reply docs/enhancements/soft-install-timeout.md:133]: Actually it was not handled

Comment (carbonin) [reply docs/enhancements/soft-install-timeout.md:120]: So do we collect the must-gather at the 24 hour time out or not at all?

Comment (ori-amizur) [reply docs/enhancements/soft-install-timeout.md:120]: If 24h timeout expires - the installation is marked as failed.  So it is handled the same way as failed installations are handled today.
The question is if we should collect must-gather that soft timeout expired and the user cancelled the installation.

**Pull request merged**